src/kw_include.sh

```diff
@@ -42,14 +43,19 @@ function include()
     return 1 # EPERM
   fi
 
-  varname="$(get_varname "$filepath")"
+  fullpath="$(realpath "$filepath")"
+  varname="$(get_varname "$fullpath")"
 
-  if [[ -v "${varname}" ]]; then
-    return 0
-  fi
+  if [[ -v KW_INCLUDES_SET ]]; then
+    [[ -v KW_INCLUDED_PATHS["$fullpath"] ]] && return 0
 
-  declare -g "${varname}"=1
+    KW_INCLUDED_PATHS["$fullpath"]=1
+  else
+    declare -g KW_INCLUDES_SET=1
+    declare -gA KW_INCLUDED_PATHS=(["$fullpath"]=1)
+  fi
 
+  declare -g "$varname"=1
```

Comment:
With this new implementation, do we even need `varname`?

Reply:
You have a really good point :) I think we don't!

Reply:
@kwy95 what tests should we keep in this new implementation? I thought of only keeping these three:

![image](https://user-images.githubusercontent.com/39812919/142047749-4024ccb4-f36c-452b-bcf3-39a21eea6af3.png)

Reply:
I don't think we can have too many tests. I especially would like to keep the tests that deal with problems we've had before, this way we can avoid regressions in the future.

Reply:
@kwy95 any suggestions on effective name collision testing?